airflow/cli/cli_parser.py

```diff
@@ -41,11 +42,12 @@
     GroupCommand,
     core_commands,
 )
+from airflow.cli.utils import CliConflictError
 from airflow.exceptions import AirflowException
 from airflow.executors.executor_loader import ExecutorLoader
 from airflow.utils.helpers import partition
 
-airflow_commands = core_commands
+airflow_commands = core_commands.copy()  # make a copy to prevent bad interactions in tests
```

Comment:
since we do `airflow_commands.extend(...)` after that, and `airflow_commands` is just pointing to `core_commands`, we were actually modifying `core_commands`, and whatever we were adding in there stayed there for all tests.